train/uft/train.py

```diff
@@ -267,6 +269,15 @@ def train_loop(
             torch.cuda.empty_cache()
             global_step_counter += 1
 
+            early_stopping(vaL_denoiser_loss)
```

Comment:
for UFT, you also need to look at val_discriminator_loss. if both val_discriminator_loss and val_denoiser_loss are not getting better in "n" consecutive epochs, then you can stop? (i think you need to change the __call__ method of the EarlyStopping class to require two parameters, not one)

Reply:
@superleesa 
What should I set for default "n"? 3 or 5?

Reply:
@superleesa 
if we only calculate the validation loss once every epoch, isn't 3 better? Cuz the default value of epoch is 20 for sft and 10 for uft.

Reply:
In terms of patience, a higher value is more conservative than lower values. I simply thought that if we set the patience too small, the training can stop "too early" even though it still has a potential to be a better model. We can first try with 5 and then reduce to smaller value if 5 was too much.

Reply:
@superleesa 
okay, i got it